src/vmm/src/builder.rs

```diff
@@ -591,15 +605,15 @@ pub fn build_microvm(
         attach_gpu_device(
             &mut vmm,
             event_manager,
-            _shm_region,
+            gpu_shm_region,
             intc.clone(),
             virgl_flags,
             #[cfg(target_os = "macos")]
             _map_sender,
         )?;
     }
     #[cfg(not(feature = "tee"))]
-    attach_fs_devices(&mut vmm, &vm_resources.fs, None, intc.clone())?;
+    attach_fs_devices(&mut vmm, &vm_resources.fs, dax_shm_region, intc.clone())?;
```

Comment:
I have the feeling, from what I have tested, that this share memory region can't be used by more than one virtio device.

Reply:
Seems to work fine for me? I don't see why it wouldn't work as long as the range used by each device does not overlap (which is why I split it in half here).

Reply:
Yes, the half/half splitting between gpu and fs clearly works.
If the user has two virtiofs devices, they end up sharing the exact same portion, though.
That makes all my virtiofs devices fail to setup, except for the first one.

Reply:
Ohh, sorry, I misunderstood. Maybe @slp knows what the intent was here? I only brought back the previous behavior from back when DAX was enabled, so I'm guessing this same problem already existed before?

Reply:
We need to implement some kind of shm region manager that manages the window and can be used to request regions of a certain size. I'm adding this to my TODO list and I think I'll have bandwidth to implement it next week, in case no one beats me implementing it first.